File: 프로그래머스/1/12934. 정수 제곱근 판별/정수 제곱근 판별.py
import math

def isInteger(sqrt):
    floor = int(math.floor(sqrt));
    
    if sqrt - floor == 0: 
        return True
    else :
        return False
    

def solution(n):
    answer = 0
    
    sqrt = math.sqrt(n);
    
    if isInteger(sqrt):
        answer = math.pow(sqrt + 1, 2)
    else :
        answer = -1;
        
    return answer

File: 프로그래머스/1/12934. 정수 제곱근 판별/정수 제곱근 판별.java
class Solution {
    
    private boolean isInt(Number number) {
        
        double sqrt = Math.sqrt((double) number);
        int intValue = (int) sqrt;
        
        return sqrt - intValue > 0 ? false : true;
    }
    
    public long solution(long n) {
        long answer = 0;
        long sqrt = 0;
        
        if (isInt((double) n)) {
            sqrt = (long) Math.sqrt(n);
            answer = (long) Math.pow(sqrt + 1 , 2);
        }
        else {
            answer = -1;
        }
        
        return answer;
    }
}